src/compiler/checker.ts

```diff
@@ -6420,14 +6440,14 @@ namespace ts {
             }
         }
 
-        function containsArgumentsReference(declaration: FunctionLikeDeclaration): boolean {
+        function containsArgumentsReference(declaration: SignatureDeclaration): boolean {

```

Comment:
Was there a specific need to change the type? Introducing the cast on line 6450 seems to indicate the original type here was correct.